### PR TITLE
Shallow water/updating python scripts

### PR DESCRIPTION
**Pull request by miguelmaso**
The Stage and the solvers are derived classes from the base classes in the core

### CODE REVIEW

File: applications/MeshingApplication/custom_processes/multi_scale_refining_process.cpp [hunk 1]
```diff
@@ -0,0 +1,963 @@
...
+    mEchoLevel = mParameters["echo_level"].GetInt();
+    mDivisionsAtLevel = mParameters["number_of_divisions_at_level"].GetInt();
+
+    mOwnName = mParameters["own_model_part_name"].GetString();
+    mRefinedName = mParameters["refined_model_part_name"].GetString();
+
+    mElementName = mParameters["element_name"].GetString();
+    mConditionName = mParameters["condition_name"].GetString();
+
+    std::string own_name = mParameters["own_model_part_name"].GetString();
```
Comment (loumalouomega) [inline applications/MeshingApplication/custom_processes/multi_scale_refining_process.cpp:59]: const std::string& ?

Comment (loumalouomega) [reply applications/MeshingApplication/custom_processes/multi_scale_refining_process.cpp:59]: Idem for the other strings

File: applications/MeshingApplication/custom_processes/multi_scale_refining_process.cpp [hunk 2]
```diff
@@ -0,0 +1,963 @@
...
+    std::string refined_name = mParameters["refined_model_part_name"].GetString();
+
+    mInterfaceName = mParameters["refining_interface_model_part"].GetString();
+    mInterfaceConditionName = mParameters["refining_boundary_condition"].GetString();
+
+    // Get the model part hierarchy
+    StringVectorType sub_model_parts_names;
+    if (mrRootModelPart.HasSubModelPart(own_name))
```
Comment (loumalouomega) [inline applications/MeshingApplication/custom_processes/multi_scale_refining_process.cpp:67]: Be careful with this comments, maybe use {}

File: applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py [hunk 3]
```diff
@@ -1,104 +1,124 @@
...
     def AddDofs(self):
         raise Exception("Calling the base class instead of the derived one")
 
+    def ImportModelPart(self):
+        # we can use the default implementation in the base class
+        self._ImportModelPart(self.main_model_part,self.settings["model_import_settings"])
+
+    def PrepareModelPart(self):
+        # Defining the variables
+        gravity = self.settings["gravity"].GetDouble()
```
Comment (loumalouomega) [inline applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:78]: don't you prefer to set the gravity with a process?

**Review by loumalouomega — APPROVED** [review]
Some minor comments

**Comment by loumalouomega** [comment]
Conflicts!!!!!

File: applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py (hunk 3 above)
Comment (miguelmaso) [reply applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:78]: there is something strange

Comment (miguelmaso) [reply applications/ShallowWaterApplication/python_scripts/shallow_water_base_solver.py:78]: sorry, it is the wrong branch

**Pull request closed without merging**